File: src/test/java/fr/sogilis/blog/article/kotlinobjectmother/KingdomMother.java
package fr.sogilis.blog.article.kotlinobjectmother;

public class KingdomMother {

    public static KingdomBuilder one() {
        return new KingdomBuilder()
            .name("Logres")
            .capital(TownMother.one().build());
    }

    public static KingdomBuilder logres() {
        return one()
            .name("Logres")
            .capital(TownMother.kaamelott().build());
    }
}

File: src/test/kotlin/fr/sogilis/blog/article/kotlinobjectmother/KingdomMother.kt
package fr.sogilis.blog.article.kotlinobjectmother

object KingdomMother {

    fun one() = KingdomBuilder().apply {
        name = "Logres"
        capital = TownMother.one().build()
    }

    fun logres() = one().apply {
        name = "Logres"
        capital = TownMother.kaamelott().build()
    }
}


class KingdomBuilder : Builder<Kingdom> {
    var name: String? = null
    var capital: Town? = null

    override fun build() = Kingdom(name!!, capital!!)
}
